frontends/verilog/verilog_parser.y

```diff
@@ -1933,10 +1951,32 @@ typedef_decl:
 			rewriteAsMemoryNode(astbuf1, $5);
 		}
 		addTypedefNode($4, astbuf1); }
-	| TOK_TYPEDEF non_wire_data_type type_name ';'   { addTypedefNode($3, $2); }
+	| TOK_TYPEDEF enum_struct_type type_name ';'   { addTypedefNode($3, $2); }
 	;
 
-non_wire_data_type:
+data_type:
```

Comment:
This production differs greatly from `data_type` in the official grammar. I'd prefer we call this something like `typdef_base_type` to make the distinction clear.